test_cases/stdlib/builtins/check_exception_group.py

```diff
@@ -0,0 +1,327 @@
+from __future__ import annotations
+
+import sys
+from typing import TypeVar
+from typing_extensions import assert_type
+
+if sys.version_info >= (3, 11):
+    # BaseExceptionGroup
+    # ==================
+
+    # `BaseExceptionGroup` can work with `BaseException`:
+    beg = BaseExceptionGroup("x", [SystemExit(), SystemExit()])
+    assert_type(beg, BaseExceptionGroup[SystemExit])
+    assert_type(beg.exceptions, tuple[SystemExit | BaseExceptionGroup[SystemExit], ...])
+
+    # Covariance works:
+    _beg1: BaseExceptionGroup[BaseException] = beg
+
+    # `BaseExceptionGroup` can work with `Exception`:
+    beg2 = BaseExceptionGroup("x", [ValueError()])
+    # FIXME: this is not right, runtime returns `ExceptionGroup` instance instead,
+    # but I am unable to represent this with types right now.
+    assert_type(beg2, BaseExceptionGroup[ValueError])
+
+    # .subgroup()
+    # -----------
+
+    assert_type(beg.subgroup(KeyboardInterrupt), BaseExceptionGroup[KeyboardInterrupt] | None)
+    assert_type(beg.subgroup((KeyboardInterrupt, SystemExit)), BaseExceptionGroup[BaseException] | None)
+
+    def is_base_exc(exc: BaseException) -> bool:
+        return isinstance(exc, BaseException)
+
+    def is_specific(exc: SystemExit | BaseExceptionGroup[SystemExit]) -> bool:
+        return isinstance(exc, SystemExit)
+
+    # This one does not have `BaseExceptionGroup` part,
+    # this is why we treat as an error.
+    def is_system_exit(exc: SystemExit) -> bool:
+        return isinstance(exc, SystemExit)
+
+    def unrelated_subgroup(exc: KeyboardInterrupt) -> bool:
+        return False
+
+    assert_type(beg.subgroup(is_base_exc), BaseExceptionGroup[SystemExit] | None)
+    assert_type(beg.subgroup(is_specific), BaseExceptionGroup[SystemExit] | None)
+    beg.subgroup(is_system_exit)  # type: ignore
+    beg.subgroup(unrelated_subgroup)  # type: ignore
+
+    # This might be strange at first, `ValueError` is not listed in the generic params.
```

Comment:
Not sure I understand this comment, the type doesn't seem strange to me.

Reply:
Yes, it was outdated. Replaced.